### PR TITLE
Eng 16120

### PR DESCRIPTION
**Pull request by manjujames**

### CODE REVIEW

File: src/frontend/org/voltdb/DRLogSegmentId.java [hunk 1]
```diff
@@ -98,8 +113,12 @@ public static boolean isInitialAckDRId(long drId) {
         return false;
     }
 
-    public static boolean seqIsBeforeZero(long drId) {
-        return (getSequenceNumberFromDRId(-1L) == getSequenceNumberFromDRId(drId));
+    public static boolean seqIsBeforeZero(long drId, byte clusterId) {
+        if (drId == -1 && clusterId == 0) {
```
Comment (JadenW) [inline src/frontend/org/voltdb/DRLogSegmentId.java]: Is this change necessary?  Previously method still works? 
Is the change for making seqIsBeforeZero() method more consistent with isInitialAckDRId() check?

Comment (manjujames) [reply src/frontend/org/voltdb/DRLogSegmentId.java]: This was used from stats call. The implementation was incorrect and hence converting max sequence numbers to -1.

File: src/frontend/org/voltdb/DRLogSegmentId.java [hunk 2]
```diff
@@ -73,6 +73,11 @@ public static long makeInitialAckDRId(int clusterId) {
         return makeDRIdFromComponents(clusterId, 0L) - 1L;
     }
 
+    public static boolean isMaxDrIdForCluster(long drId, byte clusterId) {
```
Comment (rdykiel) [inline src/frontend/org/voltdb/DRLogSegmentId.java]: Any synchronization issues here?

File: src/frontend/org/voltdb/DRLogSegmentId.java [hunk 3]
```diff
@@ -86,6 +91,16 @@ public static boolean isEmptyDRId (long drId) {
         return ((drId >>> 63) == 1L) && (drId != -1);
     }
 
+    // Constructing InitialAckDRId changes the clusterId in the DrId.
+    // To correctly say if a DRId is an initial ack id, we need the clusterId as well.
+    public static boolean isInitialAckDRId(long drId, byte clusterId) {
+        if (drId == -1 && clusterId == 0) {
+            return true;
+        }
+        drId += 1;
```
Comment (rdykiel) [inline src/frontend/org/voltdb/DRLogSegmentId.java]: So if it was -1 it's now 0, right? Is there a chance this is being called with drId == -1 and clusterId != 0?

File: src/frontend/org/voltdb/DRLogSegmentId.java (hunk 2 above)
Comment (manjujames) [reply src/frontend/org/voltdb/DRLogSegmentId.java]: No, it is just looking at the 2 passed in parameters.

File: src/frontend/org/voltdb/DRLogSegmentId.java (hunk 3 above)
Comment (manjujames) [reply src/frontend/org/voltdb/DRLogSegmentId.java]: Yes. In that case we want it to return false.
